htdocs/comm/propal/class/propal.class.php

```diff
@@ -694,7 +694,7 @@ public function addline($desc, $pu_ht, $qty, $txtva, $txlocaltax1 = 0.0, $txloca
 				$result = $product->fetch($fk_product);
 				$product_type = $product->type;
 
-				if (getDolGlobalString('STOCK_MUST_BE_ENOUGH_FOR_PROPOSAL') && $product_type == 0 && $product->stock_reel < $qty) {
+				if ($product->stockable_product && $product->stock_reel < $qty && getDolGlobalString('STOCK_MUST_BE_ENOUGH_FOR_PROPOSAL') && ($product_type == Product::TYPE_PRODUCT || ($product_type == Product::TYPE_SERVICE && getDolGlobalString('STOCK_SUPPORTS_SERVICES')))) {
```

Comment:
I suggest to replace with
`if ($product->isStockManaged() && $product->stock_reel < $qty && getDolGlobalString('STOCK_MUST_BE_ENOUGH_FOR_PROPOSAL')) {`

Reply:
You are right, I changed the PR this way.